google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateSubscriptionAndPullMessages.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.examples.pubsub.snippets;
+
+import com.google.cloud.pubsub.spi.v1.MessageReceiver;
+import com.google.cloud.pubsub.spi.v1.Subscriber;
+import com.google.cloud.pubsub.spi.v1.SubscriberClient;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.common.util.concurrent.Service;
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.PushConfig;
+import com.google.pubsub.v1.SubscriptionName;
+import com.google.pubsub.v1.TopicName;
+
+/**
+ * A snippet for Google Cloud Pub/Sub showing how to create a Pub/Sub pull subscription and
+ * asynchronously pull messages from it.
+ */
+public class CreateSubscriptionAndPullMessages {
+
+  public static void main(String... args) throws Exception {
+    TopicName topic = TopicName.create("test-project", "test-topic");
+    SubscriptionName subscription = SubscriptionName.create("test-project", "test-subscription");
+
+    try (SubscriberClient subscriberClient = SubscriberClient.create()) {
+      subscriberClient.createSubscription(subscription, topic, PushConfig.getDefaultInstance(), 0);
+    }
+
+    MessageReceiver receiver =
+        new MessageReceiver() {
+          @Override
+          public ListenableFuture<MessageReceiver.AckReply> receiveMessage(PubsubMessage message) {
+            System.out.println("got message: " + message.getData().toStringUtf8());
+            return Futures.immediateFuture(MessageReceiver.AckReply.ACK);
+          }
+        };
+    Subscriber subscriber = null;
+    try {
+      subscriber = Subscriber.Builder.newBuilder(subscription, receiver).build();
+      subscriber.addListener(
+          new Service.Listener() {
+            @Override
+            public void failed(Service.State from, Throwable failure) {
+              System.err.println(failure);
+            }
+          },
+          MoreExecutors.directExecutor());
+      subscriber.startAsync().awaitRunning();
```

Comment:
These primitives are provided by the [Service class](https://google.github.io/guava/releases/snapshot/api/docs/com/google/common/util/concurrent/Service.html). We are essentially making this synchronous though since we `awaitRunning` directly after.